### PR TITLE
MET-4270 Fixing publish links malfunction for failed datasets

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java [hunk 1]
```diff
@@ -41,197 +43,205 @@
...
+
+        // collect steps processing information
+        List<ProgressByStepDto> stepsInfo = new LinkedList<>();
+        recordsProcessedByStep.forEach((step, statusMap) -> addStepInfo(stepsInfo, statusMap, step,
+                recordErrorsByStep));
+
+        return new ProgressInfoDto(
+                getPublishPortalUrl(dataset, completedRecords),
+                dataset.getRecordsQuantity(), completedRecords,
+                stepsInfo, datasetInfoDto, getErrorMessage(completedRecords, failedRecords, dataset));
```
Comment (jeortizquan) [inline src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java:122]: my suggestion is: ```getErrorMessage(completedRecords, failedRecords, dataset.getRecordsQuantity())``` to keep variables at the same level.

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java [hunk 2]
```diff
@@ -41,197 +43,205 @@
...
+            return PROCESSING_DATASET_MESSAGE;
+        }
+
+        var datasetId = dataset.getDatasetId() + SEPARATOR + dataset.getDatasetName() + SUFFIX;
+        return portal + URLEncoder.encode(datasetId, StandardCharsets.UTF_8);
     }
-    if (!recordsQty.equals(completedRecords)) {
-      return PROCESSING_DATASET_MESSAGE;
+
+    private String getErrorMessage(Long completedRecords, Long failedRecords, DatasetEntity dataset) {
```
Comment (jeortizquan) [inline src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java:143]: my suggestion is: ```private String getErrorMessage(Long completedRecords, Long failedRecords, Long recordsQty) ``` and you can remove the line ```Long recordsQty = dataset.getRecordsQuantity();```

File: src/test/java/eu/europeana/metis/sandbox/dto/report/ProgressInfoDtoTest.java [hunk 3]
```diff
@@ -133,6 +133,6 @@ private static ProgressInfoDto getTestCompletedInfoDto() {
             Language.HR,
             Country.CROATIA,
             false,
-            false));
+            false), "");
   }
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/dto/report/ProgressInfoDtoTest.java:137]: my suggestion is to add the following unit test:
```
 @NotNull
  private static ProgressInfoDto getTestErroTypeInfoDto() {
    return new ProgressInfoDto("http://metis-sandbox",
        5L,
        5L,
        getProgressByStepDtoList(5, 5),
        new DatasetInfoDto("datasetId",
            "datasetName",
            LocalDateTime.parse("2022-03-14T22:50:22"),
            Language.HR,
            Country.CROATIA,
            false,
            false), "Error");
  }
  
  @Test
  void getErrorType() {
    progressInfoDto = getTestErroTypeInfoDto();
    assertEquals( "Error", progressInfoDto.getErrorType());
    assertEquals( "", progressInfoDto.getPortalPublishUrl());
  }
```

**Review by jeortizquan — CHANGES_REQUESTED** [review]
There are suggestions to change to keep same level concept of variables that are being used and a unit test.

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java (hunk 1 above)
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java:122]: Done

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java (hunk 2 above)
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetReportServiceImpl.java:143]: Done

File: src/test/java/eu/europeana/metis/sandbox/dto/report/ProgressInfoDtoTest.java (hunk 3 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/dto/report/ProgressInfoDtoTest.java:137]: Added

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**